VIRTUAL_VACATION/vacation.pl

```diff
@@ -332,9 +356,67 @@ sub check_for_vacation {
     return $rv;
 }
 
+#
+# Get Accountname stored in name of table mailbox
+#
+sub get_accountname {
+    my ($from_mailbox) =@_;
+    my $logger = get_logger();
+
+    my $query = qq{SELECT name FROM mailbox WHERE username=? };
+    my $stm = $dbh->prepare($query) or panic_prepare($query);
+    $stm->execute($to) or panic_execute($query,"username='$from_mailbox'");
+    my @row = $stm->fetchrow_array;
+    my $rv = $stm->rows;
+
+    my $accountname = $row[0];
+
+    return $accountname;
+}
+
+#
+# Replace <%From_Date>   with date part of activefrom out the vacation table on base of email
+# Replace <%Until_Date>  with date part of activeuntil out the vacation table on base of email
+#
+# The varibale $replace_form and $replace_until will have the <%From_Date> and <%Until_Date> replace text
+
+sub replace_string {
+    my ($to) =@_;
+    my $logger = get_logger();
+
+    my $query = qq{SELECT body,activefrom,activeuntil FROM vacation WHERE email=? };
+    my $stm = $dbh->prepare($query) or panic_prepare($query);
+    $stm->execute($to) or panic_execute($query,"email='$to'");
+    my @row = $stm->fetchrow_array;
+    my $rv = $stm->rows;
 
-# try and determine if email address has vacation turned on; we
-# have to do alias searching, and domain aliasing resolution for this.
+    my $vacation_body = $row[0];
+    my $f_date = substr($row[1],0,10);
```

Comment:
why do you need substr calls here?
row[1] should be a database timestamp field, shouldn't it?